packages/containers/src/List/List.container.js

```diff
@@ -204,28 +216,37 @@ class List extends React.Component {
 			}
 		}
 
+		props.list.cellDictionary = {
+			...connectedCellDictionary,
+		};
+
 		if (props.cellDictionary) {
-			props.list.cellDictionary = Object.keys(props.cellDictionary).reduce((acc, key) => {
-				const current = props.cellDictionary[key];
-				// eslint-disable-next-line no-param-reassign
-				acc[key] = {
-					...omit(current, ['component']),
-					cellRenderer: props.getComponent(current.component),
-				};
-				return acc;
-			}, {});
+			props.list.cellDictionary = {
+				...props.list.cellDictionary,
+				...Object.keys(props.cellDictionary).reduce(
+					(accumulator, key) => ({
```

Comment:
You don't need to create a new object at each iteration. We are dealing with object that we just created, so we can modify them.

```
const cellDictionary = { ...connectedCellDictionary };
if (props.cellDictionary) {
	Object.keys(props.cellDictionary).forEach(key => {
		const current = props.cellDictionary[key];
		// eslint-disable-next-line no-param-reassign
		cellDictionary[key] = {
			...omit(current, ['component']),
			cellRenderer: props.getComponent(current.component),
		};
	});
}

props.list.cellDictionary = cellDictionary;
```